src/mame/namco/namcos10.cpp

```diff
@@ -1111,6 +1383,20 @@ void namcos10_memm_state::ns10_mrdrilr2(machine_config &config)
 ///////////////////////////////////////////////////////////////////////////////////////////////
 // MEM(N)
 
+void namcos10_memn_state::pio_dma_write(uint32_t *p_n_psxram, uint32_t n_address, int32_t n_size)
+{
+	// Are DMA writes ever performed?
+	logerror("%s: pio_dma_write: unhandled DMA write %08x %08x\n", machine().describe_context(), n_address, n_size);
+}
+
+void namcos10_memn_state::pio_dma_read(uint32_t *p_n_psxram, uint32_t n_address, int32_t n_size)
+{
+	uint16_t *ptr = reinterpret_cast<uint16_t*>(&p_n_psxram[n_address / 4]);
+	for (int32_t i = 0; i < n_size * 2; i++) {
+		ptr[i] = nand_data_r();
+	}
```

Comment:
You need to use `auto ptr = little_endian_cast<uint16_t>(&p_n_psxram[n_address / 4]);` (from src/lib/util/endianness.h) to avoid endianness issues.